generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -1395,4 +1396,50 @@ public boolean preconditionHolds(Expr expr, BasicType basicType, boolean constCo
           && !constContext;
     }
   }
+
+  /**
+   * Identity transformation to multiply a vector or a rectangular matrix P by an identity matrix I,
+   * producing the same vector/matrix P as output. This relies on two properties of the identity
+   * matrix:
+   *     If P is a matrix of size n x m, where n is the number of columns and m the number of rows,
+   *     and I is the identity matrix of size n x n, then P * I = P.
+   *     If P is a matrix of size n x m, where n is the number of columns and m the number of rows,
+   *     and I is the identity matrix of size m x m, then I * P = P.
+   *     Note that matrices of size 1 x m or n x 1 are vecm or vecn, respectively.
+   * When performed, transforms an expression of a vector or rectangular matrix P ->
+   * identity(identityMatrix * P) or identity(P * identityMatrix).
+   */
+  private class IdentityMatrixMultIdentity extends AbstractIdentityTransformation {
+    private IdentityMatrixMultIdentity() {
+      // We want all non-square matrices and floating point vectors.
```

Comment:
".filter(item -> P(item))" keeps only those elements that *satisfy* P.  Looks like you've written this assuming the opposite.

I suggest you pass the union of BasicType.allNonSquareMatrixTypes and VEC2, VEC3, VEC4.

Reply:
Oops, you're exactly right. I just figured out that Streams can be concatenated, so I'll use that instead.